### PR TITLE
I6916 updates to get firefox btn on non firefox client (on guides pages)

### LINKED ISSUE #6916: Follow up on how +Add for guide pages should look/be implemented on non-firefox browsers
See https://github.com/mozilla/addons-frontend/issues/6432#issuecomment-433083079

I am adding this as its own issue so it stays on our radar.

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #6916 

## Desktop
Alt text

## Mobile
Alt text

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Thanks for this, @rebmullin. Part of your approach actually addresses a different bug (#6985), which I plan on addressing separately. I'm going to do that right away, so you may want to wait for that and rebase your patch after it lands.

**Comment by bobsilverberg** [comment]
@rebmullin #6985 has landed which addresses the `AddonCompatibilityError` portion of your patch.

**Comment by willdurand** [comment]
I see this on mobile:

screen shot 2018-11-20 at 17 57 33

File: src/amo/components/GuidesAddonCard/styles.scss [hunk 1]
```diff
@@ -13,6 +13,14 @@
     }
   }
 
+  .GetFirefoxButton--guides.GetFirefoxButton {
+    font-size: $font-size-default;
+    max-width: 170px;
+    min-width: 160px;
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/components/GuidesAddonCard/styles.scss]: Note: I removed both `max-width` and `min-width` and it works quite well on all screens.

[the remark this answers is not in the document]
Comment (rebmullin) [reply src/amo/components/GuidesAddonCard/styles.scss]: I tried width and without these but they were different sizes on some cards:

Alt text

[the remark this answers is not in the document]
Comment (rebmullin) [reply src/amo/components/GuidesAddonCard/styles.scss]: I guess I can remove this and see if UX/design says anything..?

File: src/amo/components/GuidesAddonCard/index.js [hunk 2]
```diff
@@ -153,20 +149,12 @@ export class GuidesAddonCardBase extends React.Component<InternalProps> {
...
-                  {
-                    utm_content: addon.guid,
-                  },
-                )}`}
-                puffy
-                className="Button--get-firefox"
-              >
-                {i18n.gettext('Only with Firefox—Get Firefox Now')}
-              </Button>
+            {addon && (
```
Comment (willdurand) [inline src/amo/components/GuidesAddonCard/index.js]: Is this still needed given that L105 there is already this condition?

File: src/amo/components/GuidesAddonCard/styles.scss [hunk 3]
```diff
@@ -49,6 +49,12 @@
   @include respond-to(medium) {
     flex-wrap: nowrap;
   }
+
+  .GetFirefoxButton {
+    font-size: $font-size-default;
+    text-align: center;
+    white-space: normal;
+  }
 }
```
Comment (willdurand) [inline src/amo/components/GuidesAddonCard/styles.scss]: This is a general comment, not something to change here: it'd be nice to only have one CSS block per main CSS class name and not some in `.GuidesAddonCard` and some below.

File: src/amo/components/GuidesAddonCard/index.js (hunk 2 above)
Comment (willdurand) [reply src/amo/components/GuidesAddonCard/index.js]: I believe you can remove it.

Comment (rebmullin) [reply src/amo/components/GuidesAddonCard/index.js]: good point

**Pull request merged**